agent/util/git.go

```diff
@@ -0,0 +1,37 @@
+package util
+
+import (
+	"bytes"
+	"os/exec"
+	"strings"
+
+	"github.com/evergreen-ci/evergreen/thirdparty"
+	"github.com/pkg/errors"
+)
+
+// getGitVersion retrieves the current Git version installed on the system.
+func getGitVersion() (string, bool, error) {
+	cmd := exec.Command("git", "--version")
+	var out bytes.Buffer
+	cmd.Stdout = &out
+	err := cmd.Run()
+	if err != nil {
+		return "", false, errors.Wrap(err, "failed to get git version")
+	}
+
+	return thirdparty.ParseGitVersion(strings.TrimSpace(out.String()))
+}
+
+// IsGitVersionMinimum checks if the installed Git version is later than the specified version.
+func IsGitVersionMinimum(minVersion string) (bool, error) {
```

Comment:
This helper is quite neat but it's doing two things: 1. checking git version 2. checking if it's apple. The func itself doesn't state that and it (looking at the comment below) it looks specific to Scalar. SInce we only use it with Scalar, could you make this func in to `IsGitVersionMinimumForScalar()` maybe?

Also, could you put a link or docs as to where you found it wasn't bundled with Apple just for comments purposes?

Reply:
I changed it to IsGitVersionMinimumForScalar. I didn't find official enough docs on it, I happened to run into the error and figured it out.